### PR TITLE
text to hooks

### PR DESCRIPTION
**Pull request by talesdsp**

### CODE REVIEW

File: src/Text.js [hunk 1]
```diff
@@ -1,8 +1,7 @@
-import React, { Component } from "react";
-import { Animated, Text, StyleSheet } from "react-native";
-
+import React from "react";
+import { Animated, StyleSheet, Text } from "react-native";
 import expoTheme from "./theme";
-import { spacing, parseSpacing, mergeTheme } from "./utils";
+import { d, mergeTheme, parseSpacing } from "./utils";
```
Comment (talesdsp) [inline src/Text.js]: the 'spacing' name is used 2 times on this component. The one imported from ./utils is not recognized but the one declared from props. I think we should rename it something else

File: src/Text.js [hunk 2]
```diff
@@ -116,138 +115,133 @@ class Typography extends Component {
...
+    transform,
+    regular,
+    bold,
+    semibold,
+    medium,
+    weight,
+    light,
+    center,
+    right,
+    spacing, // letter-spacing
```
Comment (talesdsp) [inline src/Text.js:140]: the second

File: src/Text.js (hunk 1 above)
Comment (hetmann) [reply src/Text.js]: We can import Utils as `import * as Utils from './utils';` and in this way we can have exclusivity on the component Props.

And `spacing` can be renamed `getSpacing` from Utils.
I do have 2 methods named `getMargins` and `getPaddings` that do the same thing  as `spacing`, `spacing` method was added for abstractization the other 2 methods.

File: src/utils/index.js [hunk 3]
```diff
@@ -1,19 +1,13 @@
...
-export {
-  rgba,
-  getMargins,
-  getPaddings,
-  spacing,
-  parseSpacing,
-  mergeTheme,
-  theme
-};
+const d = spacing;
```
Comment (hetmann) [inline src/utils/index.js]: Maybe `getSpacing` is a better name instead of `d`, is telling the user what to expect when using this method.

**Review by hetmann — APPROVED** [review]

File: src/utils/index.js (hunk 3 above)
Comment (talesdsp) [reply src/utils/index.js]: Sure! That was just to debug 😄

**Pull request merged**